pkg/model/universe.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package model
+
+import (
+	"io/ioutil"
+	"strings"
+
+	"github.com/gobuffalo/flect"
+
+	"sigs.k8s.io/kubebuilder/pkg/scaffold/input"
+	"sigs.k8s.io/kubebuilder/pkg/scaffold/resource"
+	"sigs.k8s.io/kubebuilder/pkg/scaffold/util"
+)
+
+// Universe describes the entire state of file generation
+type Universe struct {
+	Project *input.ProjectFile
+
+	Boilerplate string `json:"boilerplate,omitempty"`
+
+	Resource *Resource `json:"resource,omitempty"`
+
+	Files []*File `json:"files,omitempty"`
+
+	// Multigroup tracks if the project has more than one group
+	MultiGroup bool `json:"multigroup,omitempty"`
+}
+
+// NewUniverse creates a new Universe
+func NewUniverse(options ...UniverseOption) (*Universe, error) {
+	universe := &Universe{}
+
```

Comment:
```suggestion
universe.Boilerplate = "-"
```
For by default it be just with  `"-"` and then, we be able to simplify by removing the func WithoutBoilerplate

WDYT?

Reply:
There is a difference between using the functional option `WithoutBoilerplate` and not using it. I'm not sure if it is applied in this PR or will be in a following one (Im on my phone now and can't check it) but if you leave `Boilerplate` as `""`, `Scaffold.Execute` will try to find the Boilerplate in the default path but if you set it to anything else it will not overwrite it. That's why `WithoutBoilerplate` sets it to `"-"` (any value different to the empty string would work).